x/tbr/internal/keeper/keeper.go

```diff
@@ -0,0 +1,186 @@
+package keeper
+
+import (
+	"github.com/commercionetwork/commercionetwork/x/tbr/internal/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/exported"
+)
+
+type Keeper struct {
+	StoreKey sdk.StoreKey
+
+	BankKeeper         bank.Keeper
+	StakeKeeper        staking.Keeper
+	DistributionKeeper distribution.Keeper
+
+	Cdc *codec.Codec
+}
+
+func NewKeeper(storeKey sdk.StoreKey, bk bank.Keeper, sk staking.Keeper, dk distribution.Keeper, cdc *codec.Codec) Keeper {
+	return Keeper{
+		StoreKey:           storeKey,
+		BankKeeper:         bk,
+		StakeKeeper:        sk,
+		DistributionKeeper: dk,
+		Cdc:                cdc,
+	}
+}
+
+//Utility method to set Block Reward Pool
+func (k Keeper) SetBlockRewardsPool(ctx sdk.Context, updatedPool sdk.DecCoins) {
+	store := ctx.KVStore(k.StoreKey)
+	store.Set([]byte(types.BlockRewardsPoolPrefix), k.Cdc.MustMarshalBinaryBare(&updatedPool))
+}
+
+//Return the Block Rewards Pool if exists
+func (k Keeper) GetBlockRewardsPool(ctx sdk.Context) sdk.DecCoins {
+	var brPool sdk.DecCoins
+	store := ctx.KVStore(k.StoreKey)
+	brpBz := store.Get([]byte(types.BlockRewardsPoolPrefix))
+	k.Cdc.MustUnmarshalBinaryBare(brpBz, &brPool)
+	return brPool
+}
+
+//Increase the Block Rewards Pool with the specified coin amount
+func (k Keeper) IncrementBlockRewardsPool(ctx sdk.Context, funder sdk.AccAddress, amount sdk.Coins) {
```

Comment:
I think this method should probably return errors when: 

1. The pool does not have enough funds
2. Something goes wrong while moving the amount from the user wallets to the pool

I think you can take inspiration from [`x/accreditations/internal/keeper.go#DepositIntoPool`](https://github.com/commercionetwork/commercionetwork/blob/riccardo/user-accreditation/x/accreditations/internal/keeper/keeper.go#L77) method

Reply:
how does the pool funds compromise this method, we're only adding funds to it, not subtracting, so there will be no problem right?

Reply:
btw now i'll return an error if something goes wrong while subtracting coins from user

Reply:
I personally think that, if the `bk.HasCoins` returns `false`, we should return an error. Otherwise, completing without any error might signal the user that the fund has been approved even if no coins have been moved from its wallet into the pool. Does it sound good?